### PR TITLE
Play nicely with react-hot-loader

### PR DESCRIPTION
**Pull request by pugnascotia**
react-hot-loader's component proxying was breaking direct comparisons with the `Redirect` component type. Fix this using a suggested approach from https://github.com/gaearon/react-hot-loader/issues/304.

### CODE REVIEW

**Comment by JannesV** [comment]
Is this getting merged anytime soon ?

**Comment by pugnascotia** [comment]
Rebased on latest master.

**Comment by peter-mouland** [comment]
I'm not part of this project, but you might have a better chance of it being merged if there were some tests?

**Comment by rlueder** [comment]
For those landing here with the same issue related to `react-hot-loader` breaking, an easy workaround is described here: https://github.com/reach/router/issues/153#issue-357988475

TLDR; use "path" instead of "from" in the `Redirect` component: `<Redirect path="/" to="yourpath" />`

**Comment by joelgallant** [comment]
More conservatively, this could do the "static member" strategy, if there's any fear of breakage here.

Just ran in to a production bug with this, where the warning/error did not show up in development.
